LoopKit/DeviceManager/PumpManager.swift

```diff
@@ -113,9 +113,9 @@ public protocol PumpManager: DeviceManager {
     func removeStatusObserver(_ observer: PumpManagerStatusObserver)
     
     /// Fetch the pump data (reservoir/events) if it is out of date.
-    /// After a successful fetch, the PumpManager should trigger a loop by calling the delegate method `pumpManagerRecommendsLoop(_:)`
-    func assertCurrentPumpData()
-
+    /// After a successful fetch, the PumpManager should call the completion block, and trigger a loop by calling the delegate method `pumpManagerRecommendsLoop(_:)`
+    func fetchCurrentPumpData(completion: (() -> Void)?)
```

Comment:
There is already a sort of "callback" for completion of this method, and that is the delegate method `pumpManagerRecommendsLoop`.  I was going to recommend that this callback take a flag indicating if the pumpManager does recommend a Loop, but after reviewing the DIY MDT implementation, it can fire the `pumpManagerRecommendsLoop` signal outside of an assertCurrentPumpData call, so I think we should just keep it like you have it.

Reply:
Oh, and in response to the name change: "assert" was originally chosen to indicate that this method was trying to assert that pump data was current, and if it wasn't to go fetch it.  It is not expected that pumpManagers always fetch data in response to this call, but that they do what they can to ensure that it is current.  Doing nothing if the data is current is fine.

Reply:
re: already a "callback" : Yes, I noticed `pumpManagerRecommendsLoop` in the `PumpManagerDelegate`, and I also noticed that there's a `PumpManagerStatusObserver`, which actually seemed more appropriate here than passing this completion block (which would be a _third_ type of "callback", which seems weird).  However, I tried adding the class in question (`BolusEntryViewModel`) as a `PumpManagerStatusObserver` but turns out some pumps don't notify on that observer if the data doesn't change (it needs to be notified no matter what).  I don't want to add more confusion to the `PumpManager` contract.  Are you still ok with this `completion` passed in?

re: the "assert" prefix: It's just that "assert" has so much baggage that it appears like it is `assert()`, which is not the intent.  The "fetch" prefix at least matches the comment, but I can understand if that is still not quite right.  How about `ensureCurrentPumpData()`?

Reply:
Yes, ensureCurrentPumpData works great.

Reply:
The comment states that PumpManager should call pumpManagerRecommendsLoop unequivocally. The PumpManager should only call pumpManagerRecommendsLoop if it has an accurate and up to date understanding of insulin delivery and has reported that.